### PR TITLE
Minor updates to protocol semantics

### PR DESCRIPTION
**Pull request by ilevkivskyi**
This updates protocol semantic according to the discussion in https://github.com/python/typing/pull/464:
* ``None`` should be considered a subtype of an empty protocol
* ``method = None`` rule should apply only to callable protocol members
* ``issublcass()`` is prohibited for protocols with non-method members.

Fixes https://github.com/python/mypy/issues/3906
Fixes https://github.com/python/mypy/issues/3938
Fixes https://github.com/python/mypy/issues/3939

### CODE REVIEW

**Comment by gvanrossum** [comment]
@JukkaL Can you review this?

@ilevkivskyi What changed so that reveal_type() of `builtins.list` now comes out as `builtins.list[Any]`?

**Comment by ilevkivskyi** [comment]
> What changed so that reveal_type() of ``builtins.list`` now comes out as ``builtins.list[Any]``?

This:
```
 def visit_type_type(self, t: TypeType) -> None:
-    pass
+    t.item.accept(self)
```
This fix will anyway happen in #3952, but without it this test (expectedly) crashed.

File: mypy/checkexpr.py [hunk 1]
```diff
@@ -273,6 +273,16 @@ def visit_call_expr(self, e: CallExpr, allow_none_return: bool = False) -> Type:
                         not tp.type_object().runtime_protocol):
                     self.chk.fail('Only @runtime protocols can be used with'
                                   ' instance and class checks', e)
+        if (isinstance(e.callee, RefExpr) and len(e.args) == 2 and
+                e.callee.fullname == 'builtins.issubclass'):
+            for expr in mypy.checker.flatten(e.args[1]):
```
Comment (JukkaL) [inline mypy/checkexpr.py]: Move the body to a separate method, as this function is already quite long. Maybe also move the body of the above if statement (from line 269) to a separate method. You could also move the if statement to be nested under the above if statement (line 267). You'd only need the check `if e.callee.fullname == 'builtins.subclass'` which would simplify things slightly.

**Comment by JukkaL** [comment]
Can you also fix the merge conflict?

**Comment by ilevkivskyi** [comment]
Thanks! I addressed CR and fixed the conflict.

**Pull request merged**